broker/src/main/java/io/zeebe/broker/clustering/atomix/storage/snapshot/AtomixSnapshotStorage.java

```diff
@@ -142,8 +147,13 @@ public void onNewSnapshot(
     metrics.incrementSnapshotCount();
     observeSnapshotSize(snapshot);
 
-    LOGGER.debug("Purging snapshots older than {}", snapshot);
-    store.purgeSnapshots(snapshot);
+    try {
+      LOGGER.debug("Purging snapshots older than {}", snapshot);
+      store.purgeSnapshots(snapshot);
+      DbSnapshotStore.cleanUpTemporarySnapshots(pendingDirectory, snapshot.index());
```

Comment:
Doesn't purgeSnapshot does the cleanup?

Reply:
It now only does for the one pending directory it knows about - do you think it should also purge this one as well? I'm also not sure if it's a good idea for it do it anyway, because `purgeSnapshots` can be called from different thread contexts, so we might still run into the issue of deleting on going replications? Though it should be more fault tolerant

Reply:
Sorry, I missed that DbSnapshot is shared between raft and atomix. Then it make sense to cleanup here. But then wouldn't we cleanup raft-pending snapshot while we purge snapshots here?